tests/lax_numpy_test.py

```diff
@@ -5962,6 +5962,39 @@ def np_fun(a, i, v):
     self._CheckAgainstNumpy(np_fun, jnp_fun, args_maker)
     self._CompileAndCheck(jnp_fun, args_maker)
 
+  @jtu.sample_product(
+      shape=array_shapes,
+      dtype=jtu.dtypes.all,
+      M=range(5),
+      J=range(5),
+      axis=range(5),
+  )
+  def testPutAlongAxis(self, shape, dtype, M, J, axis):
+    if axis > len(shape):
+      self.skipTest("invalid axis")
+
+    a_shape = shape[:axis] + (M,) + shape[axis:]
+    i_shape = shape[:axis] + (J,) + shape[axis:]
+    v_shape = shape[:axis] + (J,) + shape[axis:]
```

Comment:
We should make sure to test cases where the shapes are not equal, but are broadcast-compatible.